File: examples/angular/src/app/app.component.ts
import { Component } from '@angular/core';

declare var dynamsoft: any;

@Component({
	selector: 'app-root',
	templateUrl: './app.component.html',
	styleUrls: [ './app.component.css' ]
})
export class AppComponent {
	title = 'Dynamsoft Barcode Reader';
	results = '';
	reader;

	constructor() {
		const env = this;
		dynamsoft.dbrEnv = dynamsoft.dbrEnv || {};
		dynamsoft.dbrEnv.resourcesPath = 'https://demo.dynamsoft.com/dbr_wasm/js';
		//https://www.dynamsoft.com/CustomerPortal/Portal/TrialLicense.aspx
		dynamsoft.dbrEnv.licenseKey =
			't0068NQAAADAG7KITlB55pjkzxD1rnTRhcZ/KCqVoXp6vWXmjRUbhvkCl58F+mqFnhIo1Oul/qB0moA8nA1erzTPYsb4FVLk=';
		dynamsoft.dbrEnv.bUseWorker = true;
		dynamsoft.dbrEnv.onAutoLoadWasmSuccess = function() {
			env.reader = new dynamsoft.BarcodeReader();
			document.getElementById('anim-loading').style.display = 'none';
		};
		dynamsoft.dbrEnv.onAutoLoadWasmError = function(ex) {
			alert(ex);
		};
		
		let script = document.getElementById('script');
		(<HTMLScriptElement >script).src = 'https://demo.dynamsoft.com/dbr_wasm/js/dbr-6.4.1.1.min.js';
	}

	readBarcode(): void {
		const env = this;
		let image = (<HTMLInputElement>document.getElementById('uploadImage')).files[0];
		if (!image) {
			alert('Please add an image');
			return;
		}
		this.reader
			.decodeFileInMemory(image)
			.then(function(results) {
				var txts = [];
				for (var i = 0; i < results.length; ++i) {
					txts.push(results[i].BarcodeText);
				}
				env.results = JSON.stringify(txts);
			})
			.catch((ex) => {
				alert(ex);
			});
	}
}

File: example/web/angular-dart/lib/app_component.dart
@JS()
library app_component;

import 'package:angular/angular.dart';

import 'dart:html';
import 'package:js/js.dart';
import 'dart:js';
import 'dart:convert';

@JS('console.log')
external void log(dynamic str);

@JS("JSON.stringify")
external String stringify(obj);

@JS()
class Promise<T> {
  // external Promise(void executor(void resolve(T result), Function reject));
  external Promise then(Function(T result));
}

@JS("dynamsoft.BarcodeReader")
class DynamsoftBarcodeReader {
  external factory DynamsoftBarcodeReader();
  external Promise<JsArray> decodeFileInMemory(File file);
}

@Component(
  selector: 'my-app',
  styleUrls: ['app_component.css'],
  templateUrl: 'app_component.html',
)

class AppComponent implements OnInit {
  
  final title = 'Angular Dart - Barcode Reader';
  String results = '';
  DynamsoftBarcodeReader reader;

  void ngOnInit() async {
    results = '';
    reader = new DynamsoftBarcodeReader();
  }

  void getResults(JsArray barcodes) {
    int len = barcodes.length;
    var json = jsonDecode(stringify(barcodes));
    var tmp = '';
    for (int i = 0; i < len; i++) {
      tmp += json[i]['BarcodeText'] + '; ';
    }
    
    SpanElement spanElement= document.querySelector('#barcodeResults');
    spanElement.text = tmp;
  }

  void readBarcode() {
    InputElement input = document.querySelector('#uploadImage');
    FileList files = input.files;
    File file = files.item(0);

    if (reader != null) {
      reader.decodeFileInMemory(file).then(
        allowInterop(getResults)
      );
    } else {
      results = file.name;
    }
  }
  
}


